joinmarket-webui/joinmarket-webui
# Improve lock/unlock mechanism

Closes #157.

Adaptions:
- Display "There can be only one active wallet" only when multiple wallets are available
- Make alerts on wallet page are not dismissable: Once dismissed it would hide potential upcoming alerts.
- Show currently active wallet as "Active" (even if it is not "your" session)
- Sort wallets by name - always show active wallet first
- Display taker/maker indicator for active wallet
- Display dialog before locking wallet if maker/taker is running, with option to "just logout" (remove token)
- Remove message "This wallet is active, but there is no token to interact with it. Please remove the lock file on the server." completely.

Some of these adaptions might not be good ideas. Please feel free to suggest changes that come to mind. e.g. Why show a modal in the first place? If a maker is running, cannot the app decide to just logout by itself?

How to test the "confirm lock modal":
- Start maker or collaborative transaction
- Try to lock wallet
- Modal should be present with option to either confirm "lock" or use suggested action "logout"
  - "lock": Will lock the wallet and stop taker/maker operation
  - "logout": will forget the local auth token - wallet will still be displayed as "active" with an activity indicator



#### Wallet screen of active wallet with running maker (without token)
Notice: Wallet with active session is now displayed as "Active"

##### Before/After
<img src="https://user-images.githubusercontent.com/3358649/158827293-da0769d2-9790-4cc5-acd6-9950ae4cf30b.png" width="300" /> | <img src="https://user-images.githubusercontent.com/3358649/158827878-8374e571-976b-4696-b96c-3e6d024b448a.png" width="300" />



#### Wallet screen of active wallet with running maker (with token)
Notice: Missing message - it is now only shown when multiple wallets exist

##### Before/After
<img src="https://user-images.githubusercontent.com/3358649/158827669-2a598bf7-9779-4129-9eed-1c2360e8f297.png" width="300" /> | <img src="https://user-images.githubusercontent.com/3358649/158827978-d2f90a53-6a73-42d4-8fa8-ea2c70019488.png" width="300" />

#### Wallet screen of multiple wallets (without token)
Notice: Wallets are ordered by name
##### Before/After
<img src="https://user-images.githubusercontent.com/3358649/158827425-f58144b9-33d7-44b1-b58d-df48752e3261.png" width="300" /> | <img src="https://user-images.githubusercontent.com/3358649/158828031-33a18065-e11a-4043-8e4e-63deb90d2927.png" width="300" />

#### Wallet screen of multiple wallets (with token)
Notice: Active wallet always on top

##### Before/After
<img src="https://user-images.githubusercontent.com/3358649/158827755-fd21675e-f95f-4fc7-a27a-17a9bfdd1034.png" width="300" /> | <img src="https://user-images.githubusercontent.com/3358649/158828079-aad067ab-9d2d-47e8-8f88-1bca02a6b230.png" width="300" />


#### Press "lock" button on Wallet screen with taker running (with token)
<img src="https://user-images.githubusercontent.com/3358649/158829743-a2b56a56-bb2e-4470-8c96-a343c8377d7e.png" width="300" />

#### Wallet screen with taker running (without token) - after "logging out"
<img src="https://user-images.githubusercontent.com/3358649/158829795-f5e94469-92e9-4f09-9a93-fc15c45704ac.png" width="300" />

#### Press "lock" button on Wallet screen with maker running (with token)

<img src="https://user-images.githubusercontent.com/3358649/158830175-2076731b-5993-4fd8-add9-b37a1acc6764.png" width="300" />
